### PR TITLE
Final group capstone - Book an Appointment

### PR DESCRIPTION
**Pull request by paulinagonzalezc**
# Final group capstone - Book an Appointment - Front-end

Our project was built with two applications, one for the back-end and one for the front-end, this is the front-end application. Here is a link to our back-end repo: https://github.com/Nelitaa/book-an-appointment-back-end

To complete the project we followed all the guidelines and requirements:

#### Basics

- You should follow the layout of [the provided design](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign). You should change only the titles, descriptions, and photos - in order to create a website about something other than motorcycles.
- Select a theme for your website - is it going to be a website for booking doctor appointments, booking online classes, or something else?

#### Features

Depending on your team size you are required to complete a set number of features. Take inventory of your team and see where you fit in the following table:

| Team size | # of required features |
| --------- | ---------------------- |
| 1-2       | 6                      |
| 3-4       | 9                      |
| 5         | 11                     |

If your team has finished the required features, but you'd still like to complete more you are free to continue with the optional ones.

**Note: If your team decides to implement optional features, make sure they work as expected (just like the required ones). This will be considered in your final evaluation!**

##### Core features - Required for _every_ team size

1. The user logs in to the website, only by typing the username (a proper authenticated login is a requirement if your group is made of 5 people).
2. In the navigation panel, the user can see links to:
   - Motorcycles/doctors/classes/items that you selected as a theme.
   - "Reserve" form.
   - "My reservations".
   - "Add motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
   - "Delete motorcycle/doctor/class/item that you selected as a theme" (the link is visible to everybody).
3. On the main page, the user can see a [list of motorcycles/doctors/classes/items that you selected as a theme](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005577).

4. When the user selects a specific item, they can see the [details page](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005579) with its full description (skip the "Rotate image" button).

   - In the details page, the user can click the "Reserve" button (in the design you can see the "Configure" button - please replace it with the "Reserve" button).

5. When the user clicks the "Add item" link in the navigation panel they can see a form for adding a new item.

6. Make the app responsive, creating both mobile and desktop versions.

##### Additional features - Required for teams with 3-4 & 5 members (optional for others)

7. When the user clicks the "Delete item" link in the navigation panel they can see a list of all items with title and "Delete" button.
   - When the user clicks the "Delete" button, the selected item is marked as removed and does not show on the main list anymore.

- To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
  - Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
  - The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.

8. When the user clicks the "My reservations" link in the navigation panel they can see a list of their reservations (with information about item name, date and city).
9. Add full documentation for your API.

##### Additional features - Required _only_ for teams with 5 members (optional for others)

10. Implement proper user authentication from the front-end to the server.
11. Make sure that the "Add item" and "Delete item" links are accessible only by users who are admins.

#### Technical set up

- Set up the repository/repositories on GitHub and use Gitflow.
- You should use Postgres as your database
- Use Rails to create backend API.
- Use React & Redux to create frontend UI.
- You can choose if you want to set up your project as two separate apps or as one.
- The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import AddDoctor from './components/AddDoctor';
+import Reservations from './pages/Reservations';
+import DoctorDetails from './pages/DoctorDetails';
+import DeleteDoctor from './components/DeleteDoctor';
+import AddReservation from './pages/AddReservation';
+import { fetchDoctors } from './redux/doctors';
+import { fetchReservations } from './redux/reservations';
+import Redirect from './pages/Redirect';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js:18]: - [ ]  Please, kindly refactor this function to ES6 arrow functions, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will comply with best practices. 💪

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  const dispatch = useDispatch();
+  // const doctors = useSelector((state) => state.doctors);
```
Comment (Meltrust) [inline src/App.js:20]: - [ ] Kindly, remove this old commented code in order to improve the repo's cleanliness. 👍

File: src/pages/Redirect.js
Comment (Meltrust) [inline src/pages/Redirect.js:1]: - [ ] OPTIONAL: If you are not going to implement this page's content, I suggest removing it from the project in order to improve the UX, and prevent confusion. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
_note: The student is reporting that they ended up being a 2 person team.  Reviewing accordingly, as requested._

Hi @paulinagonzalezc,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  App is working well✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 

### Even if you had a troublesome team experience, all the requirements have been met,  except for the general "reserve" page.  However, reservations can still be performed by clicking on a doctor.  For a 2 person team, you did a great job, congratulations.

#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.js (hunk 1 above)
Comment (paulinagonzalezc) [reply src/App.js:18]: Thank you @Meltrust , we will change it.

File: src/App.js (hunk 2 above)
Comment (paulinagonzalezc) [reply src/App.js:20]: Thank you @Meltrust , we will remove it.

File: src/pages/Redirect.js
Comment (paulinagonzalezc) [reply src/pages/Redirect.js:1]: Thank you @Meltrust , we will just keep it because of the missing implementation of Reserve 😅

**Review by mcrd25 — COMMENTED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Successfully implemented changes 👍 
## Optional suggestions
- [OPTIONAL] I feel as though maybe the design well rather the layout of doctors page has a disconnect with rest of application. Menu should be more accessible from this page, rather than simply the "back to doctors" button. It will make it more user friendly
- [OPTIONAL] Your front-end looks great, however do remember that: the application needs to look exactly like design with exception of photos, titles and descriptions as stated in requirements. However, i do know that's not 100% possible, but there is always room for improvement

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**